backend/src/bundles/notifications/notification.repository.ts

```diff
@@ -0,0 +1,62 @@
+import { NotificationEntity } from '~/bundles/notifications/notification.entity.js';
+import { type NotificationModel } from '~/bundles/notifications/notification.model.js';
+import { type Repository } from '~/common/types/types.js';
+
+import { type UpdateNotificationRequestDto } from './types/types.js';
+
+class NotificationRepository implements Repository {
+    private notificationModel: typeof NotificationModel;
+
+    public constructor(notificationModel: typeof NotificationModel) {
+        this.notificationModel = notificationModel;
+    }
+
+    public findById(): ReturnType<Repository['findById']> {
+        return Promise.resolve(null);
+    }
+
+    public async findAll(): Promise<NotificationEntity[]> {
+        const notifications = await this.notificationModel
+            .query()
+            .where('isRead', false)
+            .execute();
+
+        return notifications.map((it) => NotificationEntity.initialize(it));
+    }
```

Comment:
I am not sure if it is a good idea to filter notifications in method called `findAll`. Maybe, you should filter it in the service or create a separate method for it

Reply:
Its better to create a separate method